Nama : Virgie Yunita Salsabil

NIM : 21110022

Kelas : S1-SD02A

### **TUGAS 5. Vektorisasi**

1. load dataset yang diberikan menjadi dataframe
2. ubah kolom "content" menjadi vector menggunakan TFIDFVectorizer dan tampilkan hasilnya
3. Tampilkan nama fiturnya mengunakan get_features_name_out
4. Berikan penjelasan representasi output TFIDF dari sklearn
5. Upload hasilnya ke github dan kumpulkan link nya saja

In [1]:
# Instalasi luwiji
!pip install luwiji

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.0/41.0 MB 8.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.3/100.3 kB 14.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 79.3 MB/s eta 0:00:00
  Created wheel for luwiji: filename=luwiji-1.2.3-py2.py3-none-any.whl size=41032442 sha256=8cfb152d4b6de62f8982a66d120cfaf63c1a70a4223d4a3a5fe68e225a6e9486
  Stored in directory: /root/.cache/pip/wheels/49/64/e5/da119dce51f27490590d60c3df8932db315ab0454251e18955
  Created wheel for jcopml: filename=jcopml-1.2.4-py2.py3-none-any.whl size=42429 sha256=2cd59860b10d7281c46af6a2ea29be70bd37d5f1b5ebc9a3ccb842d0a4751702
  Stored in directory: /root/.cache/pip/wheels/60/8e/5c/369ea57808ee92f880e0ed18b86004a0d469e9fcdee1b867d7
Successfully built luwiji jcopml


In [38]:
import numpy as np
import pandas as pd

# Import modul os untuk operasi sistem
import os

# Modul illustration untuk ilustrasi dalam pemrosesan teks
from luwiji.text_proc import illustration

# Modul word_tokenize untuk memecah teks menjadi kata-kata
from nltk.tokenize import word_tokenize

# Modul tqdm.auto untuk menampilkan bar progres di lingkungan interaktif
from tqdm.auto import tqdm

# Modul Word2Vec untuk membuat model Word2Vec
from gensim.models import Word2Vec

# Import beberapa modul dari paket sklearn.feature_extraction.text
from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer, TfidfVectorizer

### 1. load dataset yang diberikan menjadi dataframe

In [39]:
# Input & tampilkan dataset
file = pd.read_csv('data.csv')
display(file)

Unnamed: 0                         userName  \
0             0                      Gadri Bandu   
1             1         070_Putri Azzahra Nurdin   
2             2              Novrealita Setiyana   
3             3                       auah gelap   
4             4                      Adi Nugroho   
..          ...                              ...   
995         995                       TEGUH TEAM   
996         996                         Zen Flow   
997         997                     Edward Gosal   
998         998                        Zaaa Qirr   
999         999  Andik Naufal Gaming from Nabila   

                                               content  score  
0    Saya kasih bintang tiga dulu soalnga gemenya b...      3  
1    Bagus bgtt. Dari segi story, grafik sampe puzz...      5  
2    Sejauh ini aku main aku suka banget sama model...      5  
3    Semuanya sudah sempurna tapi Sangat disayangka...      3  
4    Grafik bagus,story bagus,terutama archon quest...      5  
..                                                 ...    ...  
995  Masih bingung sih dengan cara game ini ngerend...      5  
996  game nya seru dari segi story. saran buat Hoyo...      5  
997  Game ini sangat bagus dengan map yg sangat lua...      5  
998  Sangat seru dan asik untuk dimainkan grafik hd...      5  
999  Memuat game terlalu lama . Mau masuk game aja ...      2  

[1000 rows x 4 columns]

In [40]:
# Drop kolom
teks = file.drop(['Unnamed: 0', 'userName', 'score'], axis=1)
teks

content
0    Saya kasih bintang tiga dulu soalnga gemenya b...
1    Bagus bgtt. Dari segi story, grafik sampe puzz...
2    Sejauh ini aku main aku suka banget sama model...
3    Semuanya sudah sempurna tapi Sangat disayangka...
4    Grafik bagus,story bagus,terutama archon quest...
..                                                 ...
995  Masih bingung sih dengan cara game ini ngerend...
996  game nya seru dari segi story. saran buat Hoyo...
997  Game ini sangat bagus dengan map yg sangat lua...
998  Sangat seru dan asik untuk dimainkan grafik hd...
999  Memuat game terlalu lama . Mau masuk game aja ...

[1000 rows x 1 columns]

### 2. ubah kolom "content" menjadi vector menggunakan TFIDFVectorizer dan tampilkan hasilnya

In [41]:
# Ambil 5 data dan ubah kolom "content" menjadi vektor TF-IDF
sample = teks.tail(5)  # Ambil 5 data terakhir
teks = sample["content"]

In [42]:
# Tampilkan 5 data terakhir
display(teks)

995    Masih bingung sih dengan cara game ini ngerend...
996    game nya seru dari segi story. saran buat Hoyo...
997    Game ini sangat bagus dengan map yg sangat lua...
998    Sangat seru dan asik untuk dimainkan grafik hd...
999    Memuat game terlalu lama . Mau masuk game aja ...
Name: content, dtype: object

In [43]:
# Menampilkan keseluruhan teks
array_teks = np.array(teks)
array_teks

array(['Masih bingung sih dengan cara game ini ngerender mapnya yang di render seluruh map apa sekitar playernya keknya berasa berat kalo di tempat yang penuh beranimasi kaya di sumeru tepatnya di desert',
       'game nya seru dari segi story. saran buat Hoyoverse : untuk semua world quest tolong isi voiceoever nya, cuma ini aja sih yg bikin ngantuk kalo lagi jalanin world quest',
       'Game ini sangat bagus dengan map yg sangat luas dan kita sebagai pemain bebas menjelajahi mapnya dan banyaknya teka teki dan pemandangan yang sangat bagus dan event-event menarik setiap saat',
       'Sangat seru dan asik untuk dimainkan grafik hd juga',
       'Memuat game terlalu lama . Mau masuk game aja lama d.,diperbaiki lagi . Unistal . Game membosankan cuma muter² doang ..'],
      dtype=object)

- Proses TF-IDF

In [44]:
# Inisialisasi TF-IDF Vectorizer
tfidf = TfidfVectorizer()

# Transformasi teks menjadi vektor TF-IDF
tfidf_matrix = tfidf.fit_transform(teks)

In [45]:
# Tampilkan hasil
print("Hasil TF-IDF Vectorizer untuk 5 data terakhir:")
print(tfidf_matrix)

Hasil TF-IDF Vectorizer untuk 5 data terakhir:
  (0, 17)	0.15499841873225026
  (0, 73)	0.15499841873225026
  (0, 69)	0.15499841873225026
  (0, 32)	0.15499841873225026
  (0, 6)	0.15499841873225026
  (0, 53)	0.15499841873225026
  (0, 72)	0.15499841873225026
  (0, 31)	0.125051748283774
  (0, 8)	0.15499841873225026
  (0, 7)	0.15499841873225026
  (0, 33)	0.15499841873225026
  (0, 54)	0.15499841873225026
  (0, 62)	0.15499841873225026
  (0, 1)	0.15499841873225026
  (0, 38)	0.125051748283774
  (0, 63)	0.15499841873225026
  (0, 56)	0.15499841873225026
  (0, 18)	0.619993674929001
  (0, 80)	0.250103496567548
  (0, 39)	0.125051748283774
  (0, 49)	0.15499841873225026
  (0, 27)	0.10380424746219917
  (0, 23)	0.0873234054358943
  (0, 12)	0.15499841873225026
  (0, 16)	0.125051748283774
  :	:
  (2, 23)	0.08767244046687947
  (2, 16)	0.12555158496122723
  (3, 30)	0.36265071004887
  (3, 25)	0.36265071004887
  (3, 24)	0.36265071004887
  (3, 19)	0.36265071004887
  (3, 2)	0.36265071004887
  (3, 14)	0.29258430

### 3. Tampilkan nama fiturnya mengunakan *get_features_name_out*

In [46]:
# Menggunakan metode get_feature_names_out() dari objek tfidf
# utk mendapatkan daftar nama fitur yg digunakan dlm vektor TF-IDF
feature_names = tfidf.get_feature_names_out()

# Menampilkan daftar nama fitur
print("Nama fitur:\n")
print(feature_names)

Nama fitur:

['aja' 'apa' 'asik' 'bagus' 'banyaknya' 'bebas' 'beranimasi' 'berasa'
 'berat' 'bikin' 'bingung' 'buat' 'cara' 'cuma' 'dan' 'dari' 'dengan'
 'desert' 'di' 'dimainkan' 'diperbaiki' 'doang' 'event' 'game' 'grafik'
 'hd' 'hoyoverse' 'ini' 'isi' 'jalanin' 'juga' 'kalo' 'kaya' 'keknya'
 'kita' 'lagi' 'lama' 'luas' 'map' 'mapnya' 'masih' 'masuk' 'mau'
 'membosankan' 'memuat' 'menarik' 'menjelajahi' 'muter²' 'ngantuk'
 'ngerender' 'nya' 'pemain' 'pemandangan' 'penuh' 'playernya' 'quest'
 'render' 'saat' 'sangat' 'saran' 'sebagai' 'segi' 'sekitar' 'seluruh'
 'semua' 'seru' 'setiap' 'sih' 'story' 'sumeru' 'teka' 'teki' 'tempat'
 'tepatnya' 'terlalu' 'tolong' 'unistal' 'untuk' 'voiceoever' 'world'
 'yang' 'yg']


In [47]:
# Loop melalui indeks dan data pada DataFrame 'sample_data'
for i, data_idx in enumerate(sample.index):
    # Menampilkan informasi header untuk setiap data
    print(f"---------------")
    print(f"\nDATA KE-{i + 1}: \n")
    print(f"---------------")

    # Mendapatkan vektor TF-IDF untuk data tertentu dari matriks TF-IDF
    data_tf = tfidf_matrix[i].toarray()

    # Loop melalui indeks dan nama fitur untuk menampilkan nilai TF-IDF
    for j, feature_name in enumerate(feature_names):
        # Menampilkan nilai TF-IDF untuk setiap fitur dalam data tertentu
        print(f"{feature_name}: {data_tf[0][j]:.4f}")

---------------

DATA KE-1: 

---------------
aja: 0.0000
apa: 0.1550
asik: 0.0000
bagus: 0.0000
banyaknya: 0.0000
bebas: 0.0000
beranimasi: 0.1550
berasa: 0.1550
berat: 0.1550
bikin: 0.0000
bingung: 0.1550
buat: 0.0000
cara: 0.1550
cuma: 0.0000
dan: 0.0000
dari: 0.0000
dengan: 0.1251
desert: 0.1550
di: 0.6200
dimainkan: 0.0000
diperbaiki: 0.0000
doang: 0.0000
event: 0.0000
game: 0.0873
grafik: 0.0000
hd: 0.0000
hoyoverse: 0.0000
ini: 0.1038
isi: 0.0000
jalanin: 0.0000
juga: 0.0000
kalo: 0.1251
kaya: 0.1550
keknya: 0.1550
kita: 0.0000
lagi: 0.0000
lama: 0.0000
luas: 0.0000
map: 0.1251
mapnya: 0.1251
masih: 0.1550
masuk: 0.0000
mau: 0.0000
membosankan: 0.0000
memuat: 0.0000
menarik: 0.0000
menjelajahi: 0.0000
muter²: 0.0000
ngantuk: 0.0000
ngerender: 0.1550
nya: 0.0000
pemain: 0.0000
pemandangan: 0.0000
penuh: 0.1550
playernya: 0.1550
quest: 0.0000
render: 0.1550
saat: 0.0000
sangat: 0.0000
saran: 0.0000
sebagai: 0.0000
segi: 0.0000
sekitar: 0.1550
seluruh: 0.1550
semua: 0.0000
seru: 0.

In [48]:
# Buat dataframe dari matriks TF-IDF dengan nama kolom sesuai dengan fitur kata
tfidf_df = pd.DataFrame(data=tfidf_matrix.toarray(), columns=feature_names)

# Tampilkan matriks kata
print(tfidf_df)

        aja       apa      asik     bagus  banyaknya     bebas  beranimasi  \
0  0.000000  0.154998  0.000000  0.000000   0.000000  0.000000    0.154998   
1  0.144967  0.000000  0.000000  0.000000   0.000000  0.000000    0.000000   
2  0.000000  0.000000  0.000000  0.311236   0.155618  0.155618    0.000000   
3  0.000000  0.000000  0.362651  0.000000   0.000000  0.000000    0.000000   
4  0.191178  0.000000  0.000000  0.000000   0.000000  0.000000    0.000000   

     berasa     berat     bikin  ...    tempat  tepatnya   terlalu    tolong  \
0  0.154998  0.154998  0.000000  ...  0.154998  0.154998  0.000000  0.000000   
1  0.000000  0.000000  0.179683  ...  0.000000  0.000000  0.000000  0.179683   
2  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000  0.000000   
3  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000  0.000000   
4  0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.236961  0.000000   

    unistal     untuk  voiceoever     world      y

### **Representasi output TFIDF**



- Nilai yang tinggi menunjukkan kata-kata yang relatif penting atau unik untuk data tersebut.
- Nilai yang rendah menunjukkan kata-kata yang muncul secara umum di seluruh korpus.
- Nilai nol menunjukkan kata-kata yang tidak muncul dalam data tersebut.

Sebagai contoh:
- Pada data ke-1, kata "apa" memiliki bobot 0.154998, sedangkan pada data ke-2, kata "apa" memiliki bobot 0.00 (tidak muncul).
- Pada data ke-3, kata "bagus" memiliki bobot 0.311236, menunjukkan kata tersebut relatif penting dalam data ke-3.